File: hotel-erp-frontend/src/services/customerService.js
import api from "./axios";

export const getCustomers = () => api.get("/api/customers");

export const getCustomerById = (id) => api.get(`/api/customers/${id}`);

export const createCustomer = (customer) => api.post("/api/customers", customer);

export const updateCustomer = (id, customer) => api.put(`/api/customers/${id}`, customer);

export const deleteCustomer = (id) => api.delete(`/api/customers/${id}`);

export const searchCustomers = (keyword) => api.get(`/api/customers/search?keyword=${keyword}`);

File: hotel-erp-backend/src/main/java/org/example/hotelerpbackend/service/CustomerService.java
package org.example.hotelerpbackend.service;

import org.example.hotelerpbackend.entity.Customer;
import org.example.hotelerpbackend.repository.CustomerRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CustomerService {
    private final CustomerRepository customerRepository;

    public CustomerService(CustomerRepository customerRepository) {
        this.customerRepository = customerRepository;
    }

    public Customer createCustomer(Customer customer) {
        customerRepository.findByEmail(customer.getEmail()).ifPresent(existing -> {
            throw new RuntimeException("Customer email already exists");
        });

        return customerRepository.save(customer);
    }

    public List<Customer> getAllCustomers() {
        return customerRepository.findAll();
    }

    public Customer getCustomerById(Long id) {
        return customerRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Customer not found"));
    }

    public List<Customer> searchCustomers(String keyword) {
        return customerRepository
                .findByFirstNameContainingIgnoreCaseOrLastNameContainingIgnoreCase(keyword, keyword);
    }

    public Customer updateCustomer(Long id, Customer customer) {
        Customer existing = getCustomerById(id);

        existing.setFirstName(customer.getFirstName());
        existing.setLastName(customer.getLastName());
        existing.setEmail(customer.getEmail());
        existing.setPhone(customer.getPhone());
        existing.setAddress(customer.getAddress());

        return customerRepository.save(existing);
    }

    public void deleteCustomer(Long id) {
        Customer existing = getCustomerById(id);
        customerRepository.delete(existing);
    }
}
